### DISCUSSION TITLE
Pre-processing Script vs Batch Split

### DISCUSSION THREAD

**Discussion by sanjay86alld** [nextgenhealthcare/connect#4822] [Q&A] [answered]
Hi Everyone ,

I am working on Mirth Connect and my input is EDI 834 file which I need to convert into flat file records.

The input comes in a single line so I want to split it in multiline and then use batch split to split the records into transaction and pass one transaction at a time through mirth for conversion.

Issue here is - I am doing single to multiline split via pre-processing script and then using batch split i am creating transactions that needs to be passed to mirth.

When I choose both - preprocessing script & batch split option the channel doesn't seem to work.

so-

1- Can we use both - preprocessing script & batch split in a channel ?
2-What is the order - preprocessing script runs first and then batch split OR vice versa ?
3-If we cannot use both in a channel then any other approach we can follow here ?

**Comment by tonygermano** [comment, marked as answer]
You can use both, but batch split works first.

The batch script splits the input **before** messages are created in order to generate multiple messages. The pre-processor runs on each message once they have been dispatched to the channel.
